core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -108,4 +132,89 @@ static void validateMethodNames(String serviceName, Collection<MethodDescriptor<
           "duplicate name %s", method.getFullMethodName());
     }
   }
+
+  /**
+   * Creates a new builder for a {@link ServiceDescriptor}.
+   *
+   * @since 1.1.0
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2666")
+  public static Builder newBuilder() {
```

Comment:
Guava Builders is not nearly as used as Proto Builders, so that isn't a sufficient reason to mirror the proto behavior.  (Imagine if proto builders forced required fields be in the `newBuilder()` constructor.) 

Your proposal is not *clearly* better than the existing one.  It may be, but it isn't certainly better.  Considering that there is significant precedent for not requiring it, I would prefer to leave it as is.